psycop/common/sequence_models/logger.py

```diff
@@ -18,30 +23,67 @@ def handle_wandb_folder():
         )
 
 
+@runtime_checkable
+class LoggerFactory(Protocol):
+    def __init__(
+        self,
+        save_dir: Path | str,
+        experiment_name: str,
+        run_name: Optional[str] = None,
+        offline: bool = False,
+    ) -> None:
+        ...
+
+    def get_logger(
+        self,
+    ) -> plLogger:
+        ...
+
+
 @Registry.loggers.register("wandb")
-def create_wandb_logger(
-    name: Optional[str] = None,
-    save_dir: Path | str = ".",
-    version: Optional[str] = None,
-    offline: bool = False,
-    dir: Optional[Path] = None,  # noqa: A002
-    id: Optional[str] = None,  # noqa: A002
-    anonymous: Optional[bool] = None,
-    project: Optional[str] = None,
-    prefix: str = "",
-    checkpoint_name: Optional[str] = None,
-) -> WandbLogger:
-    handle_wandb_folder()
-
-    return WandbLogger(
-        name=name,
-        save_dir=save_dir,
-        version=version,
-        offline=offline,
-        dir=dir,
-        id=id,
-        anonymous=anonymous,
-        project=project,
-        prefix=prefix,
-        checkpoint_name=checkpoint_name,
-    )
+class WandbCreator(LoggerFactory):
```

Comment:
Any reason why we don't simpy just return the logger object?

Reply:
that way we don't assume the `.get_logger` method

Reply:
The idea was to unify the `__init__` interfaces, so that swapping from `wandb` to `mlflow` was just a case of swapping which object in the registry you are pointing towards. This is much easier using a protocol than a Callable type-hint.

OTOH, it's probably a relatively rare use-case, so I'm up for either.